src/App.tsx

```diff
@@ -1,8 +1,73 @@
-import React from 'react';
+import React, { Component } from 'react';
 import './App.css';
+import { GoodsList } from './components/GoodsList';
+import { getData } from './DATA_FROM_SERVER';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+export class App extends Component {
+  state = {
+    goods: [],
+  };
 
-export default App;
+  loadGoods = () => {
+    getData()
+      .then((goods) => {
+        this.setState({ goods });
+      });
+  };
+
+  loadFiveGoods = () => {
+    getData()
+      .then((goods: StateApp['goods']) => {
```

Comment:
do not need to move StateApp to global types it needed only here
`StateApp['goods']` - do not use types like it, create Good interface

Reply:
if you specify return type in getData you do not need to specify type here at all